hack/e2e.sh

```diff
@@ -47,8 +50,47 @@ function login() {
 export VSPHERE_SERVER="${GOVC_URL}"
 export VSPHERE_USERNAME="${GOVC_USERNAME}"
 export VSPHERE_PASSWORD="${GOVC_PASSWORD}"
-export VSPHERE_DATACENTER=""
+export VSPHERE_SSH_AUTHORIZED_KEY="${VM_SSH_PUB_KEY}"
+export VSPHERE_SSH_PRIVATE_KEY="/root/ssh/.private-key/private-key"
+
+# Run the vpn client in container
+docker run --rm -d --name vpn -v "${HOME}/.openvpn/:${HOME}/.openvpn/" \
+  -w "${HOME}/.openvpn/" --cap-add=NET_ADMIN --net=host --device=/dev/net/tun \
+  gcr.io/cluster-api-provider-vsphere/extra/openvpn:latest
+
+# Tail the vpn logs
+docker logs vpn
+
+# Sleep to allow vpn container to start running
+sleep 30
+
+# Retrieve an IP to be used as the kube-vip IP
+KUBECONFIG="/root/ipam-conf/capv-services.conf"
```

Comment:
Who generate this file？

Reply:
it's mounted when using this `preset-cluster-api-provider-vsphere-e2e-config` preset https://github.com/kubernetes/test-infra/blob/28f5d976bee58965855f1b2ae9c0a9a9dd931100/config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-presubmits.yaml#L31